src/overlays/actors/ovl_En_Dt/z_en_dt.c

```diff
@@ -28,59 +95,714 @@ ActorInit En_Dt_InitVars = {
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
     { 25, 70, 0, { 0, 0, 0 } },
 };
 
-#endif
+static AnimationHeader* sAnimations[EN_DT_ANIMATION_MAX] = { &object_dt_Anim_00112C, &object_dt_Anim_0005A4,
+                                                             &object_dt_Anim_000854, &object_dt_Anim_000DA8,
+                                                             &object_dt_Anim_000BE0, &object_dt_Anim_00B500 };
+
+static u8 sAnimationModes[EN_DT_ANIMATION_MAX] = { ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_ONCE,
+                                                   ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_ONCE };
+
+// clang-format off
+static s32 sVisualStateTable[] = {
+    0, EN_DT_ANIMATION_1, EN_DT_EYE_TEXTURE_LOOK_DOWN, true,  
+    0, EN_DT_ANIMATION_2, EN_DT_EYE_TEXTURE_LOOK_DOWN, true,
+    0, EN_DT_ANIMATION_3, EN_DT_EYE_TEXTURE_LOOK_DOWN, true,  
+    0, EN_DT_ANIMATION_3, EN_DT_EYE_TEXTURE_SHOCK,     false,
+    0, EN_DT_ANIMATION_5, EN_DT_EYE_TEXTURE_SHOCK,     false, 
+    0, EN_DT_ANIMATION_4, EN_DT_EYE_TEXTURE_SHOCK,     false
+};
+// clang-format on
+
+static TexturePtr sEyeTextures[] = { gDotourEyeShockTex, gDotourEyeOpenTex, gDotourEyeClosedTex, gDotourEyeLookDownTex,
+                                     gDotourEyeSquintTex };
+
+static TexturePtr sBrowTextures[] = { gDotourEyebrowHighTex, gDotourEyebrowMidTex, gDotourEyebrowLowTex };
+
+void EnDt_Init(Actor* thisx, PlayState* play) {
+    EnDt* this = THIS;
+
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 19.0f);
+    SkelAnime_InitFlex(play, &this->skelAnime, &object_dt_Skel_00B0CC, &object_dt_Anim_00112C, this->jointTable,
+                       this->morphTable, OBJECT_DT_LIMB_MAX);
+
+    this->actor.targetMode = TARGET_MODE_6;
+    this->npcEnMuto = NULL;
+    this->npcEnBaisen = NULL;
+    Collider_InitAndSetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+
+    if ((gSaveContext.save.day == 3) && (gSaveContext.save.isNight)) {
+        EnDt_SetupFinalNightState(this, play);
+    } else {
+        s32 csId = this->actor.csId;
+        s32 i = 0;
+
+        // clang-format off
+        while (csId != CS_ID_NONE) { this->csIds[i] = csId; csId = CutsceneManager_GetAdditionalCsId(csId); i++; }
+        // clang-format on
+    }
+    this->actionFunc = EnDt_SetupCutsceneNpcs;
+}
+
+void EnDt_Destroy(Actor* thisx, PlayState* play) {
+    EnDt* this = THIS;
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+void EnDt_UpdateHeadRotate(EnDt* this) {
+    s32 deltaAngle = BINANG_SUB(this->actor.yawTowardsPlayer, this->actor.world.rot.y);
+    s32 absDeltaAngle = ABS_ALT(deltaAngle);
+
+    this->headRotTarget.y = 0;
+    if (absDeltaAngle < 20000) {
+        this->headRotTarget.y = BINANG_SUB(this->actor.yawTowardsPlayer, this->actor.world.rot.y);
+
+        if (this->headRotTarget.y > 10000) {
```

Comment:
Not sure if it is standardized, but I think once upon a time it was said we wanted angles in hex and not decimal?

Reply:
I will change it if that's what's preferred, but in this case I would like to point out that the original devs almost certainly used integers to represent these angles given the nice round numbers. 10000 is a lot easier to grok at a glance than 0x2710. Perhaps it's something the community could reconsider at a later time.

Reply:
I would agree that this particular actor probably used decimal and round numbers, however they were no where near consistent in this aspect. Some places have nice round decimal numbers while others have nice round hex numbers. We decided to prefer consistency across the whole code base, and so decided to just use Hex for all of them especially with their connection to binary angles.